### ISSUE TITLE
Visibility of post doesn't change after profile is made public 

### ISSUE THREAD

**Issue by Outlet2048** [pixelfed/pixelfed#4170]
I uploaded a new post when my profile was still set to private
After changing my profile to public, my post still remains private
So from now on there will forever be a photo missing from my profile if I ever decide to put it to public again

And that also begs the question: will new followers be able to see this post after becoming a follower?

**Comment by trwnh** [comment]
privacy is per-post, not per-profile. making your profile "private" just changes the default post privacy.

new followers will be able to see the post if their server already has the post. this depends on the server implementation, though. most servers calculate the followers locally for followers-only posts, so new followers will be able to see it there. but some softwares only deliver a post once, so new followers can only see posts going forward.

basically, on the backend, a "post" is more like an email that gets sent to the servers of all your followers. changing the privacy would require resending the post to a new audience, or otherwise getting those servers to start showing or hiding the post to their locally calculated audience.

**Comment by Outlet2048** [comment]
Thanks for the detailed reply!
Will it one day be possible to change the privacy setting of a post after it has already been posted / change all post settings at once or wouldn't that be possible with activitypub? 


Aaand if I have a private profile but one of my existing followers is on server x, does that then mean that everyone on server x gets to see my post?

**Comment by trwnh** [comment]
it's theoretically possible to change all your posts to have different privacies, but it would be like sending emails to everyone. the other servers may not know what to do with such a message, and you will have to send a separate message for each post.

if you have a follower on server x, and you send a followers-only post to server x, then server x will calculate your followers there and only show it to those followers. at least, this is the expectation. technically, the server x can do what it wants. but the instructions on the envelope say it's addressed to your followers. the server x is expected to keep track of who that includes.
